mmseg/models/backbones/vit.py

```diff
@@ -0,0 +1,342 @@
+        self.with_cp = with_cp
+        # weight init mode
+        self.weight_init = weight_init
+
+    def init_weights(self, pretrained=None):
+        if isinstance(pretrained, str):
+            logger = get_root_logger()
+            state_dict = load_checkpoint(
+                self, pretrained, strict=False, logger=logger)
+            if 'pos_embed' in state_dict.keys(
+            ) and state_dict['pos_embed'].shape != self.pos_embed.shape:
+                self.pos_embed = nn.Parameter(
+                    torch.zeros(state_dict['pos_embed'].shape))
+                logger.info(msg='Reload checkpoint')
+                load_checkpoint(self, pretrained, strict=False, logger=logger)
+                self.pos_embed = nn.Parameter(self.pos_embed[:, 1:, :])
+        elif pretrained is None:
+            normal_init(self.pos_embed)
+            for n, m in self.named_modules():
+                if isinstance(m, Linear):
+                    if n.startswith('pre_logits'):
+                        kaiming_init(m.weight, mode='fan_in')
+                        constant_init(m.bias, 0)
+                    else:
+                        if self.weight_init.startswith('jax'):
+                            xavier_init(m.weight, distribution='uniform')
+                            if m.bias is not None:
+                                if 'mlp' in n:
+                                    normal_init(m.bias, std=1e-6)
+                                else:
+                                    constant_init(m.bias, 0)
+                        else:
+                            normal_init(m.weight, std=.02)
+                            if m.bias is not None:
+                                constant_init(m.bias, 0)
+                elif self.weight_init.startswith('jax') and isinstance(
+                        m, Conv2d):
+                    kaiming_init(m.weight, mode='fan_in')
+                    if m.bias is not None:
+                        constant_init(m.bias, 0)
+                elif isinstance(m, (_BatchNorm, nn.GroupNorm, nn.LayerNorm)):
+                    constant_init(m.bias, 0)
+                    constant_init(m.weight, 1)
+        else:
+            raise TypeError('pretrained must be a str or None')
+
+    def forward(self, x):
+        _, _, h, w = x.shape
+        x = self.patch_embed(x)
+        if x.shape[1] != self.pos_embed.shape[1]:
+            # Upsample pos_embed weights if input image size > 224
+            num_pathes = self.img_size // self.patch_size
+            h, w = h // self.patch_size, w // self.patch_size
+            pos_embed = self.pos_embed.reshape(
+                1, num_pathes, num_pathes,
+                self.pos_embed.shape[2]).permute(0, 3, 1, 2)
+            pos_embed = F.interpolate(
+                pos_embed, size=[h, w], align_corners=False, mode='bicubic')
+            self.pos_embed = nn.Parameter(
+                torch.flatten(pos_embed, 2).transpose(1, 2))
```

Comment:
Updating attributes dynamically during training is not supported by PyTorch.
We may set it during load pretraining.